types/access_list_tx.go

```diff
@@ -0,0 +1,117 @@
+package types
+
+import "math/big"
+
+type TxAccessList []AccessTuple
+
+type AccessTuple struct {
+	Address     Address
+	StorageKeys []Hash
+}
+
+// StorageKeys returns the total number of storage keys in the access list.
+func (al TxAccessList) StorageKeys() int {
+	sum := 0
+	for _, tuple := range al {
+		sum += len(tuple.StorageKeys)
+	}
+
+	return sum
+}
+
+// transaction structure
+type AccessListStruct struct {
+	Nonce    uint64
+	GasPrice *big.Int
+	Gas      uint64
+	To       *Address
+	Value    *big.Int
+	Input    []byte
+	V, R, S  *big.Int
+	Hash     Hash
+	From     Address
+
+	//Type TxType
+
+	ChainID    *big.Int
+	AccessList TxAccessList
+}
+
+func (tx *AccessListStruct) transactionType() TxType { return AccessListTx }
+func (tx *AccessListStruct) chainID() *big.Int       { return tx.ChainID }
+func (tx *AccessListStruct) input() []byte           { return tx.Input }
+func (tx *AccessListStruct) gas() uint64             { return tx.Gas }
+func (tx *AccessListStruct) gasPrice() *big.Int      { return tx.GasPrice } //check:EIP2718
```

Comment:
What does this comment mean? Is there a pending work to be done?

Reply:
Nope, missed that comment there. I will remove that.